### PR TITLE
Migration v2

### PR DESCRIPTION
**Pull request by YoDaMa**
## Purpose

Second wave of samples being added to Samples Repo.

### CODE REVIEW

File: device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    set(use_mqtt ON)
+    set(use_amqp ON)
+    set(use_http ON)
+endif()
+
+set(AZUREIOT_INC_FOLDER "/usr/include/azureiot" "/usr/include/azureiot/inc")
+
+set(sample_application_c_files
+    ${PROJECT_SOURCE_DIR}/../iothub_ll_client_x509_sample.c
+    ${PROJECT_SOURCE_DIR}/../../../../tools/.certs/certs.c
```
Comment (jebrando) [inline device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt:20]: >.certs [](start = 44, length = 6)

Why did you add the cert folder as a hidden folder?

File: device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+set(sample_application_h_files
+)
+
+add_executable(${PROJECT_NAME} 
+    ${sample_application_c_files} 
+    ${sample_application_h_files}
+)
+
+add_definitions(-DSET_TRUSTED_CERT_IN_SAMPLES)
```
Comment (jebrando) [inline device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt:31]: >SET_TRUSTED_CERT_IN_SAMPLES [](start = 18, length = 27)

Why would we use this?  You will make people have to decide if they need the trusted cert or not.

File: device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt (hunk 1 above)
Comment (YoDaMa) [reply device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt:20]: because I wanted to... but it can be either.

File: device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt (hunk 2 above)
Comment (YoDaMa) [reply device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt:31]: fair. For linux I will make it default.

File: device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt (hunk 1 above)
Comment (ewertons) [reply device-client/singlethreaded/x509-authentication/linux/CMakeLists.txt:20]: better be visible

File: README.md [hunk 3]
```diff
@@ -7,41 +7,179 @@ A repository of samples for use with the [Azure IoT C Client Library](https://gi
...
+If you have an issue running a sample and you believe the error is a result of the sample code, submit a Github Issue to this repo. If you believe the error is from the C Client Library, please submit a Github issue to the [C Client Library](https://github.com/Azure/azure-iot-sdk-c/issues/new/choose).
+
+If you want to contribute to this repository, consider submitting a PR.
+
+This project has adopted the Microsoft [Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct). For more information see the Code of Conduct FAQ or contact opencode@microsoft.com with any additional questions or comments.
+
+Microsoft collects performance and usage information which may be used to provide and improve Microsoft products and services and enhance your experience. To learn more, review the privacy statement.
+
 # File Tree
 ```bash
```
Comment (ewertons) [inline README.md:26]: Yoseph, I'll leave these commetns here, but perhaps we should establish a separate new guideline for documentation. Printing the file tree seems excessive.
Other projects usually regarded as having good docs (GitHub e.g.) do not have these trees printed anywhere.

Suggestion:
- list the samples, with links to the directories
- Have a short description of what they mean/do.
- List just the subfolders of each and either link to instructions in the folder, or list the instructions of how to build/use the sample right there (preferable?).

File: device-client/multithreaded/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+##./multithreaded
+**Description: ./multithreaded is run by...**
```
Comment (ewertons) [inline device-client/multithreaded/README.md:2]: What does this mean "run by...". What is the component?

File: device-client/multithreaded/amqp-shared-methods/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+##./amqp-shared-methods
+**Description: ./amqp-shared-methods is run by...**
```
Comment (ewertons) [inline device-client/multithreaded/amqp-shared-methods/README.md:2]: Same here.

File: device-client/multithreaded/amqp-shared-methods/iothub_client_sample_amqp_shared_methods.c [hunk 6]
```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (ewertons) [inline device-client/multithreaded/amqp-shared-methods/iothub_client_sample_amqp_shared_methods.c:1]: Note: we might need to review the name "shared"

File: device-client/multithreaded/module-method-invoke/iothub_client_sample_module_method_invoke.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+#include "azure_c_shared_utility/crt_abstractions.h"
+#include "azure_c_shared_utility/threadapi.h"
+
+#include "iothub_module_client.h"
+#include "iothubtransportmqtt.h"
+
+const char* targetDevice = "[target-device-id]";  // This can also be queried at runtime be programatically retrieving the environment variable IOTEDGE_DEVICEID.
+const char* targetModule = "[target-module-id]";
+const char* targetMethodName = "[method-name]";
+const char* targetMethodPayload = "[json-payload]"; // This must be a valid json value.  *If it is a string, it must be quoted* - e.g. targetMethodPayload = "\"[json-payload]\"";
```
Comment (ewertons) [inline device-client/multithreaded/module-method-invoke/iothub_client_sample_module_method_invoke.c:19]: This is not actually json. Should be tested with other sdks (Device Explorer?) to make sure it works.

File: tools/generate_readmes.sh [hunk 8]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (ewertons) [inline tools/generate_readmes.sh:1]: Yoseph,
this is quite a long PR. It would be good to break the changes and send them incrementally. 
I'll abide to the preference of the group, but that's my suggestion.

Also, we should discuss the format of documentation for the samples.

File: tools/generate_readmes.sh [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+            echo "**To Use:**" >> README.md
+        fi
+        add_readmes
+        popd
+    done
+}
+
+add_readmes
+popd
+
```
Comment (ewertons) [inline tools/generate_readmes.sh:28]: Also, before checking in we need to make sure not only all the samples build, but also work.

**Review by ewertons — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (YoDaMa) [reply README.md:26]: yes the tree is temporary just for quick communication while the repo is being built.

File: device-client/multithreaded/README.md (hunk 4 above)
Comment (YoDaMa) [reply device-client/multithreaded/README.md:2]: these are autogenerated README.md files. Still need to fill them in.
